google/api_core/client_logging.py

```diff
@@ -0,0 +1,89 @@
+import logging
+import json
+import re
+import os
+
+_LOGGING_INITIALIZED = False
+_BASE_LOGGER_NAME = "google"
+# TODO(https://github.com/googleapis/python-api-core/issues/760): Update Request / Response messages.
+REQUEST_MESSAGE = "Sending request "
+RESPONSE_MESSAGE = "Receiving response "
+
+# TODO(https://github.com/googleapis/python-api-core/issues/761): Update this list to support additional logging fields
+_recognized_logging_fields = [
+    "httpRequest",
+    "rpcName",
+    "serviceName",
+]  # Additional fields to be Logged.
+
+
+def logger_configured(logger):
+    return (
+        logger.handlers != []
+        or logger.level != logging.NOTSET
+        or logger.propagate == False
+    )
+
+
+def initialize_logging():
+    global _LOGGING_INITIALIZED
+    if _LOGGING_INITIALIZED:
+        return
+    scopes = os.getenv("GOOGLE_SDK_PYTHON_LOGGING_SCOPE", "")
+    setup_logging(scopes)
+    _LOGGING_INITIALIZED = True
+
+
+def parse_logging_scopes(scopes):
+    if not scopes:
+        return []
+    # TODO(https://github.com/googleapis/python-api-core/issues/759): check if the namespace is a valid namespace.
+    # TODO(b/380481951): Support logging multiple scopes.
+    # TODO(b/380483756): Raise or log a warning for an invalid scope.
+    namespaces = [scopes]
+    return namespaces
+
+
+def configure_defaults(logger):
+    if not logger_configured(logger):
+        console_handler = logging.StreamHandler()
+        logger.setLevel("DEBUG")
+        logger.propagate = False
+        formatter = StructuredLogFormatter()
+        console_handler.setFormatter(formatter)
+        logger.addHandler(console_handler)
+
+
+def setup_logging(scopes=[]):
+
+    # only returns valid logger scopes (namespaces)
+    # this list has at most one element.
+    logger_names = parse_logging_scopes(scopes)
+
+    for namespace in logger_names:
```

Comment:
If called with no scopes, the user didn't tell us we need to set up logging for everything, so I think all we should do stop propagation at the base logger, like we're doing now. In fact, i think we *need* to do this, because if there is no environment variable set and we set up structured logging, then the user will start getting logs as soon as they get to this version of the library, which is a no-no.